### PR TITLE
feat(modal): add tests for modal styles, torquise button and medium t…

### PR DESCRIPTION
**Pull request by umahaea**
…emplate

Add tests for Modal
Add tests for torquise button
Add tests for medium template
Remove couple of tests in mobile inputs(focus), not needed.
@wyseguyonline @JasonLantz @udhadpa @mhomolak @StommePoes for your review.

### CODE REVIEW

File: src/main/java/elements/fixtures/buttons.html [hunk 1]
```diff
@@ -21,6 +21,10 @@
 <br><button id="cta-btn" class="pe-btn__cta">CTABtn</button>
 <input id="cta-btn-hover" class="pe-btn__cta" type="submit" value="Input type-CTA-hover">
 
+<!- turquoise -->
```
Comment (StommePoes) [inline src/main/java/elements/fixtures/buttons.html]: You need two dashes after the bang(!) symbol to make a comment.

File: src/main/java/elements/fixtures/modal.html [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        }
+        pe_ajax.send();
+    }
+</script>
+
+<button id="button-modal-with-footer" , class="pe-btn__cta_t--btn_xlarge" onclick="toggleModal('modalPortalWithFooter')">ToggleModal With Footer</button>
+<div id="modalPortalWithFooter" class="modalPortal" style="display:none;">
+    <div class="modalOverlay">
+        <div id="modal-with-footer-template" class="pe-template__static-medium modalContent" tabindex="-1" role="dialog"
+             aria-label="Modal" aria-labelledby="modalContent">
```
Comment (StommePoes) [inline src/main/java/elements/fixtures/modal.html]: With aria-label, all other labels are ignored. Thus, the aria-labelledby is useless. We can either
1) use aria-label to have the stated name of the dialog, OR
2) point to the id of some element with text using aria-labelledby to label the dialog OR
3) use aria-labelledby as in (2) but also add aria-describedby pointing to the id of a longer description (simple text/alert modals only usually).

Also aria-labelledby here is pointing to an id that doesn't exist (modalContent??)

File: src/main/java/elements/fixtures/modal.html [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            <div id="modal-header" class="modalHeader">
+                <h2 id="modal-header-title-text" class="modalHeaderText pe-title">Terms and Conditions(Basic Title)</h2>
+            </div>
+            <div id="modal-body" class="modalBody" tabindex="0">
```
Comment (StommePoes) [inline src/main/java/elements/fixtures/modal.html]: why tabindex here? should this be tabbed to? why?

File: src/main/java/elements/fixtures/modal.html [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+</div>
+
+<button id="button-modal-without-footer" class="pe-btn__primary--btn_xlarge" onclick="toggleModal('modalPortalWithoutFooter')">ToggleModal Without Footer </button>
+<div id="modalPortalWithoutFooter" class="modalPortal" style="display:none;">
+    <div class="modalOverlay">
+        <div id="modal-without-footer-template" class="pe-template__static-medium modalContent" tabindex="-1" role="dialog" aria-label="Modal"
+             aria-labelledby="modalContent">
+            <div id="modalHeader" class="modalHeader">
+                <button id="modal-close" class="modalClose pe-icon--btn" onclick="toggleModal('modalPortalWithoutFooter')">
+                    <svg class="pe-icon--remove-sm-24" focusable="false" role="img" aria-hidden="false"
```
Comment (StommePoes) [inline src/main/java/elements/fixtures/modal.html]: the aria-hidden="false" here is totally optional. if you don't have it at all, it's also false.

File: src/main/java/elements/fixtures/modal.html [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
... finibus justo,Lorem ipsum dolor sit amet, dui sapien finibus justo, vel tempus dolor tortor eu leo. dui sapien finibusdui sapien finibus justo,Lorem ipsum dolor sit amet, dui sapien finibus justo, vel tempus dolor tortor eu leo. dui sapien finibus dui sapien finibus justo,Lorem ipsum dolor sit amet, dui sapien finibus justo, vel tempus dolor tortor eu leo. dui sapien finibus nibus dui sapien finibus justo,Lorem ipsum dolor sit amet, dui sapien finibus justo, vel tempus dolor tortor eu leo. dui sapien finibus  nibus dui sapien finibus justo,Lorem ipsum dolor sit amet, dui sapien finibus justo, vel tempus dolor tortor eu leo. dui sapien finibus nibus dui sapien finibus justo,Lorem ipsum dolor sit amet, dui sapien finibus justo, vel tempus dolor tortor eu leo. dui sapien finibus nibus dui sapien finibus justo,Lorem ipsum dolor sit amet, dui sapien finibus justo, vel tempus dolor tortor eu leo. dui sapien finibus nibus dui sapien finibus justo,Lorem ipsum dolor sit amet, dui sapien finibus justo, vel tempus dolor tortor eu leo. dui sapien finibus nibus dui sapien finibus justo,Lorem ipsum dolor sit amet, dui sapien finibus justo, vel tempus dolor tortor eu leo. dui sapien finibus <p/>
```
Comment (StommePoes) [inline src/main/java/elements/fixtures/modal.html]: closing p tag is backwards

File: src/main/java/elements/fixtures/modal.html [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+<div id="modalPortalWithoutFooter" class="modalPortal" style="display:none;">
+    <div class="modalOverlay">
+        <div id="modal-without-footer-template" class="pe-template__static-medium modalContent" tabindex="-1" role="dialog" aria-label="Modal"
+             aria-labelledby="modalContent">
+            <div id="modalHeader" class="modalHeader">
+                <button id="modal-close" class="modalClose pe-icon--btn" onclick="toggleModal('modalPortalWithoutFooter')">
+                    <svg class="pe-icon--remove-sm-24" focusable="false" role="img" aria-hidden="false"
+                         aria-labelledby="_3d82fc60-2926-11e7-8bd0-375a85c4a530">
+                        <title id="_3d82fc60-2926-11e7-8bd0-375a85c4a530">close dialog</title>
+                        <use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#remove-sm-24"></use>
```
Comment (StommePoes) [inline src/main/java/elements/fixtures/modal.html]: we don't need the xmlns:xlink line anymore. It's in the sprite.

File: src/main/java/elements/fixtures/modal.html [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+            // apply focus to close button...
+            headerCloseButton ? headerCloseButton.focus() : null;
+            footerCloseButton ? footerCloseButton.focus() : null;
+
+            // if open stop page underneath from scrolling...
+            document.body.style = "overflow:hidden;";
+        }
+
+        //*****MODAL CLOSED**************
+        portal.style.display === "none" ? document.body.style = "" : null;
```
Comment (StommePoes) [inline src/main/java/elements/fixtures/modal.html]: When we close the modal, do we get scrolling back on the body element? (undo overflow:hidden?)

**Comment by umahaea** [comment]
Merging it. fixed the comments and got enough approvals.
